### PR TITLE
Autoset wave format on write based on extension

### PR DESCRIPTION
**Pull request by claudiodsf**
When writing `Stream()` or `Trace()` objects to disk, file format can
autmatically be deduced from filename extension.

Does this need a test or further modifications in documentation?

### CODE REVIEW

**Comment by claudiodsf** [comment]
Not sure about what's going on with flake8. Tests pass on my machine (flake8-2.5.4).

**Comment by QuLogic** [comment]
I think a rebase to latest master should fix the flake8 problems.

**Comment by claudiodsf** [comment]
The branch seems to be up to date with respect to current master. @megies was telling me that we should maybe merge the maintenance branch into master to fix this flake8 problem.

**Comment by QuLogic** [comment]
The maintenance branch is now merged on master, so you can rebase now.

**Comment by claudiodsf** [comment]
Rebased and force-pushed.

**Comment by megies** [comment]
Would be good to have a short test for this..

**Comment by claudiodsf** [comment]
I'll work on that later today or tomorrow.

**Comment by claudiodsf** [comment]
If all the tests go green, that's good to go from my side.

**Comment by krischer** [comment]
I reverted the commit with the old test and replaced it with a designated test that used mocked functions. This is IMHO a bit cleaner and also faster as it performs no I/O. The actual write methods are tested elsewhere so this should be comprehensive enough.

Has to be rebased to remove the two unnecessary commits before merging.

**Comment by claudiodsf** [comment]
That was totally beyond my Python-fu!

**Comment by claudiodsf** [comment]
Rebased and force-pushed.

File: obspy/core/tests/test_waveform_plugins.py [hunk 1]
```diff
@@ -425,6 +426,47 @@ def test_deepcopy(self):
...
+                    self.assertEqual(p.call_count, 1)
+
+                    # Manually specifying the format name should overwrite
+                    # this.
+                    with mock.patch("obspy.io.mseed.core._write_mseed") as p:
+                        obj.write("temp." + s, format="mseed")
+                    self.assertEqual(p.call_count, 1)
+
+        # An unknown suffix should raise.
+        with self.assertRaises(TypeError):
```
Comment (megies) [inline obspy/core/tests/test_waveform_plugins.py]: Euh, do we really raise a `TypeError`.. should be a `ValueError`. But has nothing to with this PR of course, just noticed right now.

**Comment by megies** [comment]
Good to merge, I think.

**Pull request merged**